### DISCUSSION TITLE
[Nuxt 3] [SSG] root index.vue fallbacking all routes

### DISCUSSION THREAD

**Discussion by lucassonsinlima** [nuxt/nuxt#16746] [Questions]
Hi! I've searched on issues and discussions but I dont know if I missed something, so sorry if I had.

I'm building a `nuxt@3.0.0-rc.6` app using `target: static` to generate static files of each app route. So when I run `npm run generate`, it should prerender all my routes, this is working as expected, all the built files at `.output/public` are correct but when I run an `preview` or deploy my app, we've the root `index.vue` compiled file being blinked before all routes, causing a different source code in wrong routes. I was expecting that my routes would access only their respective `html` file. Is there a way to me to achieve this behavior using SSG + Nuxt 3?

I've made this simple reproduce that is only a fresh nuxt start app added with `pages/index.vue` plus `pages/get-started/index.vue`.

https://github.com/lucassonsinlima/nuxt-reproduce

![Captura de tela de 2022-07-29 16-53-05](https://user-images.githubusercontent.com/25435624/181833956-f8e819cb-6d1c-4ad6-aa5d-f6058d006971.png)

![Captura de tela de 2022-07-29 16-53-35](https://user-images.githubusercontent.com/25435624/181834030-3390b994-972a-4612-a778-7661de331e46.png)

**Comment by chris-visser** [comment]
You could do this in your CDN. I'm not familiar with a lot of CDN's, but Cloudfront has Origin failover options that allow you to direct a viewer to another origin when your first origin returns for example a 404

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/high_availability_origin_failover.html 

This means that you should be able to direct a viewer to your index file as a fallback

**Reply by lucassonsinlima** [reply]
tyy for your reply , actually I dont want this fallback, when I navigate to some path like `/get-started`, I expect the current `html` result to be from `/get-started` and not from `/` (index), this is the issue I'm facing. I didnt find anything related to this at docs, only about `404` and `200` fallbacks, seems to be a "bug" or something not documented yet.

If you start the reproduce project and preview the result, you can see that `/get-started` source code at browser is from `/` (index), and index blinks for ~0.5s before url resolves.

**Reply by ricky11** [reply]
did you solve this?

**Reply by TheAlexLichter** [reply]
1. use `nuxi generate` and `ssr: true`
2. All non-generated sites will be rendered through fallback files including 404s etc. (https://github.com/nuxt/framework/issues/5003).

**Reply by ricky11** [reply]
@manniL it wont work with dynamic routes yes?

**Reply by TheAlexLichter** [reply]
@ricky11 It works fine with dynamic routes when they are linked and the crawler can see them *or* when you provide an array of routes you want to generate.

Fallback files also work for any kind of route.
